lib/omniauth/builder.rb

```diff
@@ -15,6 +15,10 @@ def rack14?
       Rack.release.split('.')[1].to_i >= 4
```

Comment:
It looks like this should be tightened up to check that the first component of the version number is `1`.